### ISSUE TITLE
Creatable component is not editable + bad UX 

### ISSUE THREAD

**Issue by alexandernst** [JedWatson/react-select#4982]
I'd like to apologize in advance for not using the "Discussions" feature and instead filling a bug report.

That said, I'd really like to know where is this project going in regards with it's "editability". There are multiple bug reports[[1](https://github.com/JedWatson/react-select/issues/4083)][[2](https://github.com/JedWatson/react-select/issues/4009)][[3](https://github.com/JedWatson/react-select/issues/3204)][[4](https://github.com/JedWatson/react-select/issues/2296)][[5](https://github.com/JedWatson/react-select/issues/1558)][[6](https://github.com/JedWatson/react-select/issues/1278)] as well as discussions [[7](https://github.com/JedWatson/react-select/discussions/4302)] (which, btw, is the most voted discussion by far) requesting basically the same feature: a real editable behavior.

There's the `CreatableSelect` component which allows for custom values to be inserted, but once inserted they can't be edited in any way. The only way to actually "edit" the inserted value is to delete it and type it again (really bad UX, especially infuriating if a really long value was inserted and there is a typo).

Then there is the `isMulti` prop, which allows for multiple values to be selected and/or created. But the same problem happens if/when one of the values must the edited.

Some users even tried to implement "editable tags"[[8](https://codesandbox.io/s/o4kx95rory)] (I'm guessing so that they could, somehow, inject that component inside react-select ¿?). 

None of this bugs / discussions ended up being fixed / implemented, yet a lot of people seem to be interested in such a functionality. More important, this interest doesn't seem to vanish across the years (oldest bug report I could find related to this problem is from 2016).

Please do note that I'm not trying to demand or impose anything at all. You guys (the maintainers of this project) are doing excellent job and I'm so much thankful for your effort on this. I'd just like to know if there are plans for implementing this (and if "yes" what would help you to do it) or if this is something that is not in your current priorities.

**Comment by jamesphw** [comment]
Hi, I'm curious about this too. Does anyone have any thoughts or updates? Any comments about technical feasibility?

I wouldn't mind coding a workaround, or even doing the work and creating a PR, but I'm not at all familiar with the react-select code.

I do lots of sessions with users for both training and feedback, and this is the number 1 point of UX confusion. In large part because no other UI I've ever seen behaves the way react-select does, so it's unexpected (aka confusing!) behavior for 100% of users. 

React-Select is great, but if there are alternative packages that have this feature I would love to know so I can switch.

**Comment by jamesphw** [comment]
For anyone still looking for a solution, the only option we found was to switch to a new library. [MUI ](https://mui.com/material-ui/react-autocomplete/)has excellent and more UX-friendly autocomplete component that allow users to "Create" an option.

**Comment by ifightcrime** [comment]
@jamesphw good call, that's what we ended up doing too. For anyone in the Chakra world, I will plug [chakra-react-select](https://github.com/csandman/chakra-react-select) as that's been working well for us.

**Comment by tevinch** [comment]
For the multi-value editing case, it helps to separate a tag's stable ID from its editable text. A custom `MultiValueLabel` can then offer an edit button, with a native editor immediately below the select. That keeps cursor movement and edit keys separate from the select's value-removal behavior.

I put together a small [MIT component, runnable example and tests](https://github.com/tevinch/data-shape-kit/tree/main/examples/editable-tags) for React-Select 5.10.2. It uses the public component/Creatable APIs and controlled values shaped as `{ id, text }`. Saving replaces only the matching tag's text; its ID, position and neighboring records stay intact. Cancel leaves the committed value unchanged.

The accompanying pure operation also checks the text that editing started from:

```js
const result = commitTagEdit(currentTags, {
  id: editing.id,
  originalText: editing.originalText,
  draft: editing.draft,
});
if (result.ok && result.changed) onChange(result.value);
```

This avoids overwriting a tag that changed in the parent while the editor was open. Blank/duplicate values get a visible error, and a removed tag is not recreated by an old draft. The example includes Enter/Escape handling and a composition-key guard.

This is for user-owned free-text tags. It does not rename a shared option catalog or replace the single-value autocomplete behaviors discussed in the related threads. Existing community examples and other libraries remain useful options too.
